integrated_channels/xapi/serializers.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+
+"""
+Serializers for xAPI data.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from rest_framework import serializers
+
+from enterprise.api.v1.serializers import ImmutableStateSerializer
+from enterprise.models import EnterpriseCustomerUser
+
+
+class LearnerInfoSerializer(ImmutableStateSerializer):
+    """
+    Serializer for Learner info.
+
+    Model: User
+    """
+    lms_user_id = serializers.IntegerField(source='id')
+    user_username = serializers.CharField(source='username')
+    user_email = serializers.EmailField(source='email')
+    user_country_code = serializers.CharField(source='profile.country.code')
+    user_account_creation_date = serializers.DateTimeField(source='date_joined')
+
+    enterprise_user_id = serializers.SerializerMethodField()
+    enterprise_sso_uid = serializers.SerializerMethodField()
+
+    def get_enterprise_user_id(self, obj):
+        """
+        Get enterprise user id from user object.
+
+        Arguments:
+            obj (User): Django User object
+
+        Returns:
+            (int): Primary Key identifier for enterprise user object.
+        """
+        try:
+            enterprise_learner = EnterpriseCustomerUser.objects.get(user_id=obj.id)
+        except EnterpriseCustomerUser.DoesNotExist:
+            return
+
+        return enterprise_learner.id
+
+    def get_enterprise_sso_uid(self, obj):
+        """
+        Get enterprise SSO UID.
+
+        Arguments:
+            obj (User): Django User object
+
+        Returns:
+            (str): string containing UUID for enterprise customer's Identity Provider.
+        """
+        try:
+            enterprise_learner = EnterpriseCustomerUser.objects.get(user_id=obj.id)
+        except EnterpriseCustomerUser.DoesNotExist:
+            return
+
+        return enterprise_learner.enterprise_customer.identity_provider
```

Comment:
is this meant to get the learner's id passed through SSO from the IDP, or the slug of the IDP? the field name suggests the former, but the implementation and the comment suggest the latter.

Reply:
That list comes from the enrollment data reports and the `enterprise_sso_uid` has always referred to the user id. See https://github.com/edx/edx-documentation/blob/dccdfb3ff9053967415c00c855091fd6b026d65b/en_us/enterprise_api/source/api_reference/reference.rst

Brittney is correct in that this shouldn't be pointing to the slug, but instead `enterprise_learner.get_remote_id()`